subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildTreeScopeServices.java

```diff
@@ -200,6 +209,25 @@ ModuleRepositoryCacheProvider createModuleRepositoryCacheProvider(
         return new ModuleRepositoryCacheProvider(persistentCaches, inMemoryOnlyCaches);
     }
 
+    ImmutableTransformWorkspaceServices createImmutableTransformWorkspaceServices(
+        BuildTreeScopedCacheBuilderFactory buildTreeScopedCacheBuilderFactory,
+        CrossBuildInMemoryCacheFactory crossBuildInMemoryCacheFactory,
+        FileAccessTimeJournal fileAccessTimeJournal,
+        ExecutionHistoryStore executionHistoryStore,
+        CacheConfigurationsInternal cacheConfigurations
+    ) {
+        return new ImmutableTransformWorkspaceServices(
+            buildTreeScopedCacheBuilderFactory
+                .createCrossVersionCacheBuilder(CacheLayout.TRANSFORMS.getKey())
```

Comment:
❌  I think `TRANSFORMS` should now move out of the cache layout, since it is no longer part of the Gradle user home caches. I suppose there won't be any cache cleanup either anymore; that is unfortunate.

Reply:
Cache cleanup should still happen, we have some tests for that: https://github.com/gradle/gradle/blob/1ce45aa33fe70fdafd3a424da86266960a4e00d6/subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy#L1755-L1795
so I think that is all good.

Ack, will move TRANSFORMS somewhere else.

Reply:
When does the cleanup happen? When the daemon stops? Which project cache dir would then be cleaned up?

Reply:
I am pretty sure, checking with the debugger this I see transformes are cached in:
```outputDir1 = teŝt files/ArtifactTra.Test/82lwk/.gradle/transforms-3/c88020aeb4ca986e6927374d672605bd/transformed```

Since `DefaultImmutableWorkspaceProvider.withExternalHistory` set cleanup history this should be done before build finishes via:
https://github.com/gradle/gradle/blob/47ca2bcfa14c2eae4938e5953ecbf68a89e6df57/subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCleanupServices.java#L51-L59

Reply:
Is there another cache in the `.gradle` directory that we cleanup this way? Is suppose the cleanup for it should not be in `gradleUserHomeCleanupService`, right?

Reply:
Might be related that user can set cache cleanup only from the gradle home. Not sure if we clean something more in `.gradle.`

@ghale Do you have any insights on these questions?

Reply:
Cache cleanup is normally handled when the cache is closed, whenever that occurs.  For global caches that occurs when the daemon stops, for session scoped caches, that occurs when the session is stopped.  We _do_ have the highlighted code above, which causes cache cleanup to occur at the end of a build session, even for global scoped caches, but only if the cache settings are configured to use `CleanupFrequency.ALWAYS` (the default is DAILY).  This is for things like CI tooling that need to ensure that cache cleanup has occurred before saving the cache for restoration in a future job.  Point is, that's not what is causing the cleanup.  It's being cleaned up because the cache is now being closed when the build tree scope services are closed.  

Whether we use the cleanup settings from `CacheConfigurations` is another matter.  This was intended only for caches in Gradle user home.  Now that the cache directory has been moved, we should probably just use some unconfigurable defaults for this (for now).  This might mean some reconfiguring of `DefaultImmutableWorkspaceProvider` such that we pass in the cleanup frequency and unused entries values rather than the `CacheConfigurations` object.